medical/views/res_partner.xml

```diff
@@ -10,18 +10,15 @@
         <field name="inherit_id" ref="base.view_partner_form"/>
         <field name="arch" type="xml">
             <xpath expr="//div[@name='button_box']" position="inside">
+                <field name="patient_ids" invisible="True"/>
                 <button name="medical.medical_patient_action"
                         string="Patients"
                         type="action"
                         class="oe_stat_button"
                         icon="fa-list"
-                        context="{'search_default_parent_id': active_id}"
-                        attributes="{'invisible': ['|',
-                                                   ('type', '=like', 'medical.%'),
-                                                   ('child_ids', '=', False),
-                                                   ]}"
-                        >
-                </button>
+                        context="{'search_default_id': patient_ids}"
+                        attrs="{'invisible': [('patient_ids', '=', [])]}"
+                        />
```

Comment:
It might be nice to add a domain here that filters the view to the right set of patients. That's not happening currently, and the result is confusing.